ICSharpCode.Decompiler/CSharp/ExpressionBuilder.cs

```diff
@@ -763,6 +763,21 @@ TranslatedExpression TranslateCeq(Comp inst, out bool negateOutput)
 				.WithRR(rr);
 		}
 
+		TranslatedExpression TryUniteEqualityOperandType(TranslatedExpression operand, TranslatedExpression otherOperand)
+		{
+			// Special case for enum flag check "(enum & EnumType.SomeValue) == 0"
+			// so that the const 0 value is printed as 0 integer and not as enum type, e.g. EnumType.None
+			if (operand.ResolveResult.IsCompileTimeConstant &&
+				operand.ResolveResult.Type.IsCSharpPrimitiveIntegerType() &&
+				(int)operand.ResolveResult.ConstantValue == 0 &&
```

Comment:
`operand.ResolveResult.ConstantValue` might be something else than a boxed `int`, so this can go wrong. This makes the unit tests fail. See https://ci.appveyor.com/project/icsharpcode/ilspy/build/job/wnhtrte5t1i7g5u9/tests